pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_MPSTNCuda_Measure.cpp

```diff
@@ -92,3 +92,68 @@ TEMPLATE_TEST_CASE("Probabilities", "[Measures]", float, double) {
         REQUIRE_THROWS_AS(measure.probs({2, 1}), LightningException);
     }
 }
+
+TEMPLATE_TEST_CASE("Samples", "[Measures]", float, double) {
+    using TensorNetT = MPSTNCuda<TestType>;
+
+    constexpr uint32_t twos[] = {
+        1U << 0U,  1U << 1U,  1U << 2U,  1U << 3U,  1U << 4U,  1U << 5U,
+        1U << 6U,  1U << 7U,  1U << 8U,  1U << 9U,  1U << 10U, 1U << 11U,
+        1U << 12U, 1U << 13U, 1U << 14U, 1U << 15U, 1U << 16U, 1U << 17U,
+        1U << 18U, 1U << 19U, 1U << 20U, 1U << 21U, 1U << 22U, 1U << 23U,
+        1U << 24U, 1U << 25U, 1U << 26U, 1U << 27U, 1U << 28U, 1U << 29U,
+        1U << 30U, 1U << 31U};
+
+    SECTION("Looping over different wire configurations:") {
+        // Probabilities calculated with Pennylane default.qubit:
+        std::vector<TestType> expected_probabilities = {
+            0.67078706, 0.03062806, 0.0870997,  0.00397696,
+            0.17564072, 0.00801973, 0.02280642, 0.00104134};
+
+        // Defining the State Vector that will be measured.
+        std::size_t bondDim = GENERATE(4, 5);
+        std::size_t num_qubits = 3;
+        std::size_t maxBondDim = bondDim;
+
+        TensorNetT mps_state{num_qubits, maxBondDim};
+
+        mps_state.applyOperations(
+            {{"RX"}, {"RX"}, {"RY"}, {"RY"}, {"RX"}, {"RY"}},
+            {{0}, {0}, {1}, {1}, {2}, {2}},
+            {{false}, {false}, {false}, {false}, {false}, {false}},
+            {{0.5}, {0.5}, {0.2}, {0.2}, {0.5}, {0.5}});
+        mps_state.append_mps_final_state();
+
+        auto measure = MeasurementsTNCuda<TensorNetT>(mps_state);
+
+        std::size_t N = std::pow(2, num_qubits);
+        std::size_t num_samples = 100000;
+        const std::vector<std::size_t> wires = {0, 1, 2};
+        auto samples = measure.generate_samples(wires, num_samples);
+
+        std::vector<std::size_t> counts(N, 0);
+        std::vector<std::size_t> samples_decimal(num_samples, 0);
+
+        // convert samples to decimal and then bin them in counts
+        for (std::size_t i = 0; i < num_samples; i++) {
+            for (std::size_t j = 0; j < num_qubits; j++) {
+                if (samples[i * num_qubits + j] != 0) {
+                    samples_decimal[i] += twos[(num_qubits - 1 - j)];
+                }
+            }
+            counts[samples_decimal[i]] += 1;
+        }
```

Comment:
And/or define a `sample_2_counts` function in `TestHelpers.hpp`?

Reply:
Done